src/c++/vdo/base/dedupe.c

```diff
@@ -3098,19 +3110,25 @@ int vdo_message_dedupe_index(struct hash_zones *zones, const char *name)
 	return -EINVAL;
 }
 
+void vdo_set_dedupe_state_normal(struct hash_zones *zones)
+{
+	vdo_set_admin_state_code(&zones->state, VDO_ADMIN_STATE_NORMAL_OPERATION);
+}
+
+#if defined(VDO_INTERNAL) || defined(INTERNAL)
 int vdo_add_dedupe_index_sysfs(struct hash_zones *zones)
 {
 	int result = kobject_add(&zones->dedupe_directory,
 				 &zones->completion.vdo->vdo_directory, "dedupe");
 
 	if (result == 0) {
-		vdo_set_admin_state_code(&zones->state,
-					 VDO_ADMIN_STATE_NORMAL_OPERATION);
+		vdo_set_dedupe_state_normal(zones);
 	}
```

Comment:
Can we call vdo_set_dedupe_state_normal() unconditionally at some point in the VDO load process? It feels weird that the internal code calls this necessary thing at a different time than the upstream code.